src/compiler/binder.ts

```diff
@@ -2521,7 +2521,14 @@ namespace ts {
                         // Declare a 'member' if the container is an ES5 class or ES6 constructor
                         constructorSymbol.members = constructorSymbol.members || createSymbolTable();
                         // It's acceptable for multiple 'this' assignments of the same identifier to occur
-                        declareSymbol(constructorSymbol.members, constructorSymbol, node, SymbolFlags.Property, SymbolFlags.PropertyExcludes & ~SymbolFlags.Property);
+                        if (hasDynamicName(node)) {
+                            bindAnonymousDeclaration(node, SymbolFlags.Property, InternalSymbolName.Computed);
+                            const members = constructorSymbol.assignmentDeclarationMembers || (constructorSymbol.assignmentDeclarationMembers = createMap());
+                            members.set("" + getNodeId(node), node);

```

Comment:
is this how other late-bound symbols work in the binder?

Reply:
_Other_ late bound symbols show up in their container's syntactic list of `members` - this is not so for assignments, however we still need a list of declarations to look at in the checker; so we collect that list here.